cloud/apitoken/apitoken.go

```diff
@@ -0,0 +1,260 @@
+
+	astrocore "github.com/astronomer/astro-cli/astro-client-core"
+	"github.com/astronomer/astro-cli/context"
+	"github.com/astronomer/astro-cli/pkg/ansi"
+	"github.com/astronomer/astro-cli/pkg/input"
+	"github.com/astronomer/astro-cli/pkg/printutil"
+)
+
+var (
+	ErrInvalidAPITokenKey           = errors.New("invalid ApiToken selected")
+	ErrNoAPITokenNameProvided       = errors.New("you must give your ApiToken a name")
+	ErrNoAPITokensFoundInDeployment = errors.New("no ApiTokens found in your deployment")
+	apiTokenPagnationLimit          = 100
+)
+
+func CreateDeploymentAPIToken(name, role, description, deployment string, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+
+	if name == "" {
+		fmt.Println("Please specify a name for your ApiToken")
+		name = input.Text(ansi.Bold("\nApiToken name: "))
+		if name == "" {
+			return ErrNoAPITokenNameProvided
+		}
+	}
+
+	mutateAPITokenInput := astrocore.CreateDeploymentApiTokenRequest{
+		Role:        role,
+		Name:        name,
+		Description: &description,
+	}
+	resp, err := client.CreateDeploymentApiTokenWithResponse(httpContext.Background(), ctx.Organization, deployment, mutateAPITokenInput)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "The apiToken was successfully created with the role %s\n", role)
+	return nil
+}
+
+func UpdateDeploymentAPITokenRole(apiTokenID, role, deployment string, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	var apiToken *astrocore.ApiToken
+
+	if apiTokenID == "" {
+		// Get all dep apiTokens. Setting limit to 1000 for now
+		limit := 1000
+		apiTokens, err := GetDeploymentAPITokens(client, deployment, limit)
+		if err != nil {
+			return err
+		}
+		apiToken, err = getAPIToken(apiTokens)
+		if err != nil {
+			return err
+		}
+		apiTokenID = apiToken.Id
+	} else {
+		resp, err := client.GetDeploymentApiTokenWithResponse(httpContext.Background(), ctx.Organization, deployment, apiTokenID)
+		if err != nil {
+			fmt.Println("error in GetDeploymentApiTokenWithResponse")
+			return err
+		}
+		err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+		if err != nil {
+			fmt.Println("error in NormalizeAPIError")
+			return err
+		}
+		apiToken = resp.JSON200
+	}
+
+	mutateAPITokenInput := astrocore.UpdateDeploymentApiTokenRequest{
+		Role: role,
+		Name: apiToken.Name,
+	}
+	fmt.Println("deployment: " + deployment)
+	resp, err := client.UpdateDeploymentApiTokenWithResponse(httpContext.Background(), ctx.Organization, deployment, apiToken.Id, mutateAPITokenInput)
+	if err != nil {
+		fmt.Println("error in MutateDeploymentApiTokenRoleWithResponse")
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		fmt.Println("error in NormalizeAPIError")
```

Comment:
@sunkickr @kushalmalani 
do we normally have this here and if so do we want to keep doing it like this?
I'm wondering since it would be meaningless for the user to see, but I guess it would be useful for us if we need to debug

Reply:
I think most API calls have this but I have never seen it actually come up

Reply:
Yeah would be useful for debugging. Can we improve that error message though? it looks like a debugging log

Reply:
tbh it seems unecessary to me as the method that calls it prints out a prettier error message

e.g.

`shrividyahegde@Hegde ~ % astro deployment user update --deployment-id clt6yb6vm001r01nmwzffhbs9
Enter a user Deployment role or custom role name to update user: sample

Please select the user:
 #     FULLNAME            EMAIL                             ID                            DEPLOYMENT ROLE      CREATE DATE              
 1     Shrividya Hegde     shrividya.hegde@astronomer.io     cl27lfgxr8518182mwleoycmz     DEPLOYMENT_ADMIN     2022-04-20T13:11:07Z     

> 1
error in NormalizeAPIError
Error: Invalid request: Role sample does not exist
shrividyahegde@Hegde ~ % `

Reply:
I removed them